src/main/java/io/leonis/algieba/tree/FiniteStateMachine.java

```diff
@@ -0,0 +1,46 @@
+package io.leonis.algieba.tree;
+
+import com.google.common.collect.Table;
+import java.util.function.BiConsumer;
+
+/**
+ * The Interface FiniteStateMachine.
+ *
+ * This interface describes the functionality of a finite state machine.
+ *
+ * @param <S> The type of {@link State}
+ * @author Rimon Oz
+ */
+public interface FiniteStateMachine<S extends FiniteStateMachine.State> extends Graph<S> {
+
+  /**
+   * Transitions from one state to the next, if the state transition is supported by this finite
+   * state machine.
+   *
+   * @param from The state to transition from.
+   * @param to   The state to transition to.
+   * @return True if successful, false otherwise.
+   */
+  default boolean transition(final S from, final S to) {
+    if (this.getStateTransitionTable().contains(from, to)) {
+      this.getStateTransitionTable().get(from, to).accept(from, to);
```

Comment:
If you make it a `BiFunction` instead of a `BiConsumer` you can expand functionality to not go to the next state based specific validation rules etc.

This method would also assume less side effecting:
```
if(!contains(from, to))
   return false;
return get(from, to).accept(from, to);
```